core/src/main/java/org/eclipse/hono/config/HealthCheckConfigProperties.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * ******************************************************************************
+ *  * Copyright (c) 2016, 2019 Contributors to the Eclipse Foundation
+ *  *
+ *  * See the NOTICE file(s) distributed with this work for additional
+ *  * information regarding copyright ownership.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  ******************************************************************************
+ *
+ */
+package org.eclipse.hono.config;
+
+import org.eclipse.hono.util.Constants;
+import org.eclipse.hono.util.PortConfigurationHelper;
+
+import java.util.Objects;
+
+/**
+ * Configuration of common properties that are valid for a health check server.
+ *
+ */
+public class HealthCheckConfigProperties extends AbstractConfig {
```

Comment:
In order to not introduce too much duplication of code, I would suggest to rename this class to `ServerConfig` and then let other config classes like `org.eclipse.hono.util.config.ServiceConfigProperties` extend it. We can also pull up that class' *nativeTlsRequired*, *insecurePortEnabled*, *insecurePortBindAddress* and *insecurePort* properties into this class. WDYT?

Reply:
Sounds reasonable to me.
So do we also want to use `ServerConfig` as super class of `ClilentConfigProperties`? In this case we should think about replacing `ClientConfigProperties.host` with `ServerConfig.bindAddress`. 
However I see 29 usages of `getHost()` so this would cause quite some changes. But then we also should think about the naming, as `ServerConfig` would not be proper any longer.

Reply:
> So do we also want to use ServerConfig as super class of ClilentConfigProperties?

Why would we want to do that? The `ServerConfig` properties are not relevant for clients, are they?

Reply:
At least they share `port` and `bindingAddress` (as `host`). But I guess the `insecure*` properties don't make sense in the client properties.

Reply:
As you said, these have different semantics. So we should not have `ClientConfigProperties` extend `ServerConfig`.